ignite/services/network/networkchain/init.go

```diff
@@ -101,7 +104,45 @@ func (c *Chain) initGenesis(ctx context.Context) error {
 		if err := os.WriteFile(genesisPath, genBytes, 0o644); err != nil {
 			return err
 		}
-	} else {
+
+	case c.genesisConfig != "":
+		c.ev.Send("Fetching custom Genesis from Config", events.ProgressUpdate())
+
+		// find config in downloaded source
+		path := filepath.Join(c.path, c.genesisConfig)
+		if _, err := os.Stat(path); err != nil {
+			return fmt.Errorf("the config for genesis doesn't exist: %w", err)
+		}
+
+		config, err := chainconfig.ParseFile(path)
+		if err != nil {
+			return err
+		}
+
+		// ensure that there are no validators
+		config.Validators = nil
+
+		err = c.chain.InitChain(ctx, config, false)
+		if err != nil {
+			return err
+		}
+
+		if !initAccounts {
+			err = c.chain.CheckConfigAccounts(ctx, config)
+		} else {
+			err = c.chain.InitAccounts(ctx, config)
+		}
+		if err != nil {
+			return err
+		}
+
+		// reapply since overwritten with genesis config
```

Comment:
Can we add more details in the comments for this operation?
Why is it needed?

Reply:
Sure.  This is needed because earlier, I am invoking `c.chain.InitChain()` which will populate the `.toml` files incorrectly.  We decided to not add logic that would modify those functions internally, so I just need to call the same logic again, but with a different `config` than what we are getting from the network.  

The issues we're having with this PR think that some of this related code needs refactored before this can be done cleanly